### ISSUE TITLE
NO-UNDO not specified in parameter declaration

### ISSUE THREAD

**Issue by stefandrissen** [Riverside-Software/sonar-openedge#1223]
I just updated to 3.7.2, it is throwing false positives on [define input parameter dataset](https://docs.progress.com/bundle/abl-reference/page/DEFINE-PARAMETER-statement.html) which cannot have a `no-undo`.
```
define temp-table tt no-undo
   field cc as character
   .
define dataset ds for tt.

define input parameter dataset for ds. // <-- NO-UNDO not specified in parameter declaration
```
Also for output parameter dataset:

<img width="444" height="89" alt="Image" src="https://github.com/user-attachments/assets/5b318365-27c4-4caa-b073-35cf123e6946" />

**Comment by gquerret** [comment]
Regression introduced in this version by a minor change in the syntax tree. A new version of the rules will be released soon.

**Comment by gquerret** [comment]
The riverside-rules jar file has been updated in the [existing 3.7.2 release](<https://github.com/Riverside-Software/sonar-openedge/releases/tag/V3.7.2>). Note that the previous riverside-rules package was labeled as 3.7.0, the new version is 3.7.1.
